src/arbitrary.rs

```diff
@@ -39,21 +46,69 @@ pub struct StdGen<R> {
 /// and also will specify the maximum magnitude of a randomly generated number.
 impl<R: Rng> StdGen<R> {
     pub fn new(rng: R, size: usize) -> StdGen<R> {
-        StdGen { rng: rng, size: size }
+        let pool = EntropyPool::new(rng, 4 * size);
+        let shrinker = StdShrinker::default();
+        StdGen { pool: pool,
+                 restore_buffer: Vec::new(),
+                 size: size,
+                 shrinker: shrinker,
+        }
     }
 }
 
 impl<R: Rng> Rng for StdGen<R> {
-    fn next_u32(&mut self) -> u32 { self.rng.next_u32() }
-
-    // some RNGs implement these more efficiently than the default, so
-    // we might as well defer to them.
-    fn next_u64(&mut self) -> u64 { self.rng.next_u64() }
-    fn fill_bytes(&mut self, dest: &mut [u8]) { self.rng.fill_bytes(dest) }
+    #[inline(always)]
```

Comment:
Could you remove the `inline(always)` annotations? Unless there's an observable benefit to performance, I'd prefer to leave this up to llvm.